src/cubic_loader/qlik/ods_qlik.py

```diff
@@ -97,77 +65,46 @@ def get_snapshot_dfms(table: str) -> List[DFMDetails]:
-    prefix = os.path.join(QLIK, f"{table}__ct/")
+    table_prefix = os.path.join(QLIK, f"{table}__ct/")
+    snapshot_prefix = f"{table_prefix}snapshot={etl_status.current_snapshot_ts}/"
 
-    # add archive files from snapshot folder
-    cdc_dfms: List[DFMDetails] = []
-    for dfm_file in s3_list_objects(
-        S3_ARCHIVE, f"{prefix}snapshot={etl_status.current_snapshot_ts}/", in_filter=".dfm"
-    ):
-        cdc_dfms.append(DFMDetails(path=dfm_file, ts=re_get_first(dfm_file, RE_CDC_TS)))
+    return s3_list_cdc_gz_objects(S3_ARCHIVE, snapshot_prefix, min_ts=etl_status.last_cdc_ts)
 
-    # filter error files from table folder
-    for dfm_file in s3_list_objects(S3_ERROR, prefix, in_filter=".dfm"):
-        if re_get_first(dfm_file, RE_SNAPSHOT_TS) > etl_status.current_snapshot_ts:
-            cdc_dfms.append(DFMDetails(path=dfm_file, ts=re_get_first(dfm_file, RE_CDC_TS)))
 
-    cdc_dfms = [dfm for dfm in cdc_dfms if dfm.ts > etl_status.last_cdc_ts]
-
-    return sorted(cdc_dfms, key=attrgetter("ts"))
-
-
-def thread_load_cdc_file(args: Tuple[DFMDetails, TableStatus]) -> Tuple[Optional[str], Optional[List[DFMSchemaFields]]]:
+def thread_save_csv_file(args: Tuple[str, str]) -> None:
     """
-    work to load cdc file from S3 into RDS history table
-    """
-    dfm, status = args
-    logger = ProcessLogger("load_cdc_file", dfm=dfm.path, table=status.db_fact_table)
-    try:
-        dfm_schema = dfm_schema_to_df(dfm)
-        dfm_names = ",".join(dfm_schema.get_column("name"))
-        dfm_name_set = set(dfm_schema.get_column("name"))
-
-        # check dfm schema contains CDC_COLUMNS
-        assert set(CDC_COLUMNS).issubset(dfm_name_set)
-        dfm_schema = dfm_schema.filter(pl.col("name").is_in(CDC_COLUMNS).not_())
+    work to download and partition cdc files
 
-        truth_schema = status_schema_to_df(status)
-        truth_name_set = set(truth_schema.get_column("name"))
-        new_columns = dfm_schema.join(
-            truth_schema,
-            on=truth_schema.columns,
-            how="anti",
-            join_nulls=True,
-        )
+    - download csv.gz file to tmp_folder
+    - encode header row as sha1 hash for foldername
+    - move file into hash foldername
+    """
+    csv_object, tmp_dir = args
+    logger = ProcessLogger("download_cdc_file", csv_object=csv_object)
 
-        # new_columns found, can not load csv
-        if new_columns.shape[0] > 0:
-            # check if new_columns contains columns in truth schema, would
-            # indicate that a different dimension of the table changed (type, primary key)
-            new_truth_common = truth_name_set.intersection(set(new_columns.get_column("name")))
-            assert len(new_truth_common) == 0, f"column dimension changed for {new_truth_common}"
+    try:
+        csv_local_file = csv_object.replace("s3://", "").replace("/", "|")
+        csv_local_path = os.path.join(tmp_dir, csv_local_file)
+        s3_download_object(csv_object, csv_local_path)
```

Comment:
In local testing it didn't look like this would be an issue. Limiting the size of the hash folders to ~60MB will also limit the memory usage of the dataframe that gets created from the csv files. We can keep an eye on this and tweak the `max_folder_bytes` parameter, if needed.